internal/expr/parser.go

```diff
@@ -206,17 +219,38 @@ loop:
 		switch p.input[p.pos-1] {
 		// If the preceding byte is one of these then we might be at the start
 		// of an expression.
-		case ' ', '\t', '\n', '\r', '=', ',', '(', '[', '>', '<', '+', '-', '*', '/', '|', '%':
+		case ' ', '\t', '\n', '\r', '=', ',', '[', '>', '<', '+', '-', '*', '/', '|', '%':
+			break loop
+		case '(':
+			p.bracketLevel++
+			break loop
+		case ')':
+			p.bracketLevel--
 			break loop
 		}
 	}
 
-	p.skipBlanks()
-
+	p.partStart = p.pos
 	return nil
 
 }
 
+// startSubpart is called to allow parsing of nested parts.
+func (p *Parser) startSubpart() *checkpoint {
+	scp := p.save()
+	p.init(p.input[p.pos:])
```

Comment:
As `init` feels like a method which is only called at the beginning, perhaps a comment would be good here? Perhaps something like
```go
// Clear parser state.
```
Otherwise, I think a name-change is necessary here, maybe `setInput`?

Reply:
Added comment.